test/parallel/forall/blank-intents/array-intent-from-actual-error-task.good

```diff
@@ -0,0 +1,28 @@
+array-intent-from-actual-error-task.chpl:21: error: const actual is passed to 'ref' formal 'A' of begin_fn()
+note: to formal of type [domain(1,int(64),one)] int(64)
+array-intent-from-actual-error-task.chpl:21: note: to ref formal here
+array-intent-from-actual-error-task.chpl:25: note: passed as ref here
```

Comment:
I have problems with the above error and probably with the errors below as well.

First, my understanding is that the `const A` task intent is inferred because `A` is declared `const` (on Line 21). Therefore the formal 'A' of begin_fn() should have the 'const' intent, not 'ref' intent as the error message states.

Therefore second: the error is on line 25 where a 'const' thing is assigned to. In any case the error is not on Line 21.

If this cannot be fixed, at least easily, then please explain the behavior and the circumstances (a) in the PR message and (b) in a comment in array-intent-from-actual-error-task.chpl.

If I am incorrect in my analysis, it means that the explanations are not clear enough to show the difference between the default task intent in forall vs. coforall et al.

Reply:
> First, my understanding is that the const A task intent is inferred because A is declared const (on Line 21). Therefore the formal 'A' of begin_fn() should have the 'const' intent, not 'ref' intent as the error message states.

Yes, this is because as far as the compiler is concerned, the default intent of the `begin` is still `ref-maybe-const` (in this case `ref`). So we get a const-checking error, because it passing a `const` to a `ref`, not because we are modifying the `const` (old bottom-up analysis vs new top-down analysis).

I think I can specialize the error message here specifically for the deprecation period of ref-maybe-const, and then remove it it when ref-maybe-const is removed.

Reply:
I was able to significantly improve this without much effort